feat: Added darkmode support, user and system preferences

JustOndra opened the pull request:
I have added darkmode support. 

- added user toggle dark/light to the top bar, next to the github logo
- added item view to the menu bar with Theme command which have option Light, dark and system
- fixed background in dark mode in dialog when choosing db for examples option

guyb1 commented on src/context/theme-context/theme-provider.tsx:
```diff
@@ -0,0 +1,69 @@
+        }
+    }, [theme]);
+
+    useEffect(() => {
+        if (effectiveTheme === 'dark') {
+            document.documentElement.classList.add('dark');
+        } else {
+            document.documentElement.classList.remove('dark');
+        }
+    }, [effectiveTheme]);
```
Maybe for now we should comment this part, at least until we fix the colors in dark mode.
You were right when you suggested it :)

guyb1 commented on src/pages/editor-page/top-navbar/top-navbar.tsx:
```diff
@@ -429,6 +437,37 @@ export const TopNavbar: React.FC<TopNavbarProps> = () => {
                                 </MenubarItem>
                             </MenubarContent>
                         </MenubarMenu>
+                        <MenubarMenu>
```
Also here, until we fix the colors I think it might be a good idea not to let the user get into dark mode. We should hide the buttons & menu actions that get him into dark mode.

guyb1 commented:
Hey! Great work here! @JustOndra 

I think you were right, until we fix the colors in dark mode we shouldnt let the user be in dark mode - 
we should push the toggle & the menu actions when its good for work.
I gave a few comments in the PR (remove the menu actions, toggle & comment the 'dark' class insertion)  + there are conflicts with the main branch.

Besides, it looks great! Thank you very much for your help & push!

JustOndra commented:
Yeah, I agree. @guyb1 
I update the PR as you are saying in the comments and fix the colors issue first.

You are welcome! 
I am glad I can contribute. Thank you!

guyb1 commented:
Great @JustOndra ! looking forward! :)

JustOndra commented:
Hey! @guyb1 
I fixed the PR. Commented out toggle and menu item, commented out Theme context from router. Seemed better than commenting out just useEffect.
Also I removed the bg class that makes the confict. It should be okey until I fix the colors.

guyb1 commented:
@JustOndra great work!
I merge it for now, and after the colors fix we can use it.
Thank you for your work!

guyb1 commented:
@JustOndra oh, I see there are a few conflicts. Could you solve them and I'll merge?

JustOndra commented:
@guyb1 I am sorry, I don't see any conflicts
<img width="906" alt="Snímek obrazovky 2024-08-31 v 12 20 57" src="https://github.com/user-attachments/assets/83c590c7-c227-4cd1-a1b5-c2c3b71da17f">
.

guyb1 commented:
@JustOndra Merged!
Thank you for your work here!
